src/lib/parse-purge.js

```diff
@@ -0,0 +1,62 @@
+const path = require('path');
+const fs = require('./sync-fs');
+const { warn } = require('./log');
+
+const throwError = (err) => {
+  throw new Error(`Error loading purge: ${err}`);
+};
+
+module.exports = (projectDir) => {
+  const purgePath = path.join(projectDir, 'purge.js');
+  const purgingPath = path.join(projectDir, 'purging.js');
+
+  const purgeExists = fs.exists(purgePath);
+  const purgingExists = fs.exists(purgingPath);
+
+  if (purgeExists && purgingExists) {
+    throwError(`Purge is defined at both ${purgePath} and ${purgingPath}.  Only one of these files should exist.`);
+  }
+
+  let purgeConfig;
+  if (purgeExists) {
+    try {
+      purgeConfig = require(purgePath);
+    } catch (err) {
+      warn('Expecting purge.js to module.exports={} purge configuration');
+      throwError(err);
+    }
+
+    if (purgeConfig.fn) {
+      if (typeof purgeConfig.fn !== 'function') {
+        throwError('Expecting purge.js fn export to be a function');
+      }
+      purgeConfig.fn = purgeConfig.fn.toString();
```

Comment:
Does this work when the function has dependencies from within the module?

For example: 

```
const a = 'foo';
module.exports = {
  fn: () => a === 'foo',
};
```

I can't think of an easy way to fix this sort of thing here. Although your change is a step in the right direction, I think the ideal is that we bundle the whole module in medic-conf and webapp evaluates that bundle to retrieve all the values. It's annoying for partners when valid JavaScript only partially works.

Reply:
So, I thought about this for a while and I can't find a solution that works and also doesn't require changes into how we process the purge config. :(